ConwaysGameOfLife/Generator.cs

```diff
@@ -27,19 +27,33 @@ rules checked
             universe
             rules
         */
-        Universe _universe;
+        public Universe Universe { get; set; }
+        
         IOutput _output;
 
-
-        public Generator(Universe universe, IOutput output)
+        IInput _input; 
+        public Generator(Universe universe, IOutput output, IInput input)
         {
-            _universe = universe;
+            Universe = universe;
             _output = output;
+            _input = input;
         }
 
+        public void GenerateUniverse()
+        // if lines uncommented, both tests (line 44) fail. is this method doing too much? how to test? - does it need to be tested

```

Comment:
one approach is to think of this in 2 stages - 1st stage is creating the universe with all dead cells, second stage is setting up the starter universe with live cells. lines 48 onwards can go in a separate function as it's for the second stage.